include/unifex/task.hpp

```diff
@@ -353,76 +584,181 @@ struct _awaiter {
+    // compiler to fold the members away, they must have different types. Hence,
+    // the slightly odd-looking template parameter to the empty struct.
     UNIFEX_NO_UNIQUE_ADDRESS
     conditional_t<
         needs_scheduler_t::value,
         manual_lifetime<scheduler_t>,
-        detail::_empty<0>> sched_;
+        detail::_empty<0>>
+        sched_;
     UNIFEX_NO_UNIQUE_ADDRESS
     conditional_t<
         needs_stop_token_t::value,
         inplace_stop_token_adapter<stop_token_t>,
-        detail::_empty<1>> stopTokenAdapter_;
+        detail::_empty<1>>
+        stopTokenAdapter_;
   };
 };
 
+/**
+ * The coroutine type that ensures stop requests are delivered to nested task<>s
+ * on the right scheduler.
+ */
+template <typename T>
+struct _sr_thunk_task<T>::type final : coro_holder {
+  using promise_type = typename _sr_thunk_promise<T>::type;
+  friend promise_type;
+
+private:
+  template <typename OtherPromise>
+  using awaiter = typename _awaiter<promise_type, OtherPromise>::type;
+
+  explicit type(coro::coroutine_handle<promise_type> h) noexcept
+    : coro_holder(h) {}
+
+  template <typename Promise>
+  friend awaiter<Promise>
+  tag_invoke(tag_t<unifex::await_transform>, Promise&, type&& t) noexcept {
+    return awaiter<Promise>{std::exchange(t.coro_, {})};
+  }
+};
+
+/**
+ * Await the given sa_task<> in a context that will deliver stop requests from
+ * the receiver on the expected scheduler.
+ */
+template <typename T>
+typename _sr_thunk_task<T>::type
+inject_stop_request_thunk(typename _sa_task<T>::type awaitable) {
+  // I wonder if we could do better than hopping through this extra coroutine
+  co_return co_await std::move(awaitable);
+}
+
 template <typename T>
-struct _task<T>::type : _task_base, coro_holder {
+struct _task<T>::type
+  : _task_base
+  , coro_holder {
   using promise_type = typename _promise<T>::type;
   friend promise_type;
 
-  template<
-    template<typename...> class Variant,
-    template<typename...> class Tuple>
+  template <
+      template <typename...>
+      class Variant,
+      template <typename...>
+      class Tuple>
   using value_types =
-    Variant<
-      typename std::conditional_t<std::is_void_v<T>, type_list<>, type_list<T>>
-        ::template apply<Tuple>>;
+      Variant<typename std::
+                  conditional_t<std::is_void_v<T>, type_list<>, type_list<T>>::
+                      template apply<Tuple>>;
 
-  template<
-    template<typename...> class Variant>
+  template <template <typename...> class Variant>
   using error_types = Variant<std::exception_ptr>;
 
   static constexpr bool sends_done = true;
 
+  // we can't tell whether the coroutine has any suspend points beyond the
+  // initial one and, even if we could, we wouldn't know if any of those suspend
+  // points are async
+  static constexpr blocking_kind blocking = blocking_kind::maybe;
+
+  static constexpr bool is_always_scheduler_affine = false;
```

Comment:
I realized last night that this needs to be `true`, and `with_scheduler_affinity` needs to be tweaked to invoke the *Sender's* customization if it exists, regardless of this property's value, otherwise we can't write an optimal customization of `with_scheduler_affinity` for `when_all`, which is really the motivating use case for all this churn in the first place.